modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx

```diff
@@ -0,0 +1,212 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayAlert from '@clayui/alert';
+import ClayButton from '@clayui/button';
+import {Option, Picker} from '@clayui/core';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayModal from '@clayui/modal';
+import {FieldBase} from 'frontend-js-components-web';
+import {fetch, navigate, objectToFormData} from 'frontend-js-web';
+import React, {useRef, useState} from 'react';
+
+type frontendTokenDefinitionProvider = {
+	name: string;
+	themeId: string;
+};
+
+interface AddStyleBookModalProps {
+	addStyleBookEntryURL: string;
+	closeModal: () => void;
+	frontendTokenDefinitionProviders?: Array<frontendTokenDefinitionProvider>;
+	namespace: string;
+}
+
+const AddStyleBookModalContent = ({
+	addStyleBookEntryURL,
+	closeModal,
+	frontendTokenDefinitionProviders = [],
+	namespace,
+}: AddStyleBookModalProps) =>{
+
+	const [active, setActive] = useState<boolean>(false);
+	const [loading, setLoading] = useState<boolean>(false);
+
+	const [errorMessage, setErrorMessage] = useState<string>();
+
+	const [name, setName] = useState<string>('');
+	const [themeId, setThemeId] =
+		useState<any>(frontendTokenDefinitionProviders[0]);
+
+	const formRef = useRef(null);
+
+	const handleFormError = (responseContent: any) => {
+		setErrorMessage(responseContent.error || '');
+	};
+
+	const handleSubmit = (event: any) => {
+		event.preventDefault();
+
+		const error = name.trim().length
```

Comment:
The logic for checking whether the name is added is being duplicated in multiple locations. This should be made it's own shared method.

Additionally we need to show this error directly on the field not above both fields.

Reply:
To share this logic can create function:

```
validateName(name) {
    let error = '';

    if (name.trim().length > 0) {
        error = Liferay.Language.get('this-field-is-required');
    }

    setErrorMessage(error);

    return error;
}
```

This function will need to be inside the React component since it is calling setErrorMessage which is a useState method and requires the component context.